src/librustc_monomorphize/lib.rs

```diff
@@ -0,0 +1,90 @@
+//! Monomorphization code. Formerly part of `rustc_mir`.
+
+#![feature(nll)]
+#![feature(in_band_lifetimes)]
+#![feature(box_patterns)]
+
+#![recursion_limit="256"]
+
+#![deny(rust_2018_idioms)]
+
+use rustc::{bug, traits};
+use rustc::ty::adjustment::CustomCoerceUnsized;
+use rustc::ty::{self, Ty, TyCtxt};
+use rustc::ty::query::Providers;
+use rustc::ty::Instance;
+
+pub use self::item::{MonoItem, MonoItemExt};
+
+pub mod collector;
+pub mod item;
+pub mod partitioning;
+
+pub fn provide(providers: &mut Providers<'_>) {
+    partitioning::provide(providers);
+}
+
+#[inline(never)] // give this a place in the profiler
+pub fn assert_symbols_are_distinct<'a, 'tcx, I>(tcx: TyCtxt<'a, 'tcx, 'tcx>, mono_items: I)
+    where I: Iterator<Item=&'a MonoItem<'tcx>>
```

Comment:
This looks like it should be in `rustc_codegen_utils::symbol_names` or somewhere similar?

Reply:
Moved it, as mentioned in your other comment.